dbft_test.go

```diff
@@ -0,0 +1,96 @@
+package dbft
+
+import (
+	"crypto/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/CityOfZion/neo-go/pkg/util"
+	"github.com/nspcc-dev/dbft/block"
+	"github.com/nspcc-dev/dbft/crypto"
+	"github.com/nspcc-dev/dbft/payload"
+)
+
+type Payload = payload.ConsensusPayload
+
+type testState struct {
+	myIndex    int
+	count      int
+	privs      []crypto.PrivateKey
+	pubs       []crypto.PublicKey
+	ch         chan Payload
+	currHeight uint32
+	currHash   util.Uint256
+}
+
+func TestDBFT_OnStartPrimarySendPrepareRequest(t *testing.T) {
+	const validatorsCount = 7
+
+	s := newTestState(2, validatorsCount)
+
+	t.Run("backup sends nothing on start", func(t *testing.T) {
+		s.currHeight = 0
+		service := New(s.getOptions()...)
+
+		service.Start()
+		p := tryRecv(s.ch)
+		require.Nil(t, p)
+	})
+
+	t.Run("primary send PrepareRequest on start", func(t *testing.T) {
+		s.currHeight = 1
+		service := New(s.getOptions()...)
+
+		service.Start()
+		p := tryRecv(s.ch)
+		require.NotNil(t, p)
+		require.Equal(t, payload.PrepareRequestType, p.Type())
+		require.EqualValues(t, 2, p.Height())
+		require.EqualValues(t, 0, p.ViewNumber())
+		require.NotNil(t, p.Payload())
+		require.Equal(t, s.currHash, p.PrevHash())
+		require.EqualValues(t, 2, p.ValidatorIndex())
+	})
+}
+
+func tryRecv(ch <-chan Payload) Payload {
+	select {
+	case p := <-ch:
+		return p
+	default:
+		return nil
+	}
+}
+
+func newTestState(myIndex int, count int) *testState {
+	s := &testState{
+		myIndex: myIndex,
+		count:   count,
+		ch:      make(chan Payload, 1),
+	}
+
+	s.privs, s.pubs = getTestValidators(count)
+
+	return s
+}
+
+func (s *testState) getOptions() []Option {
```

Comment:
nit: in each func you use "Test" but here not. Probably it's better to stick to one way of writing test function

Reply:
Because it is a method of a test struct.